documentation/docs/60-appendix/30-migrating-to-sveltekit-2.md

```diff
@@ -68,7 +68,7 @@ export function load({ fetch }) {
 
 ## goto(...) changes
 
-`goto(...)` no longer accepts external URLs. To navigate to an external URL, use `window.location = url`. The `state` option was removed in favor of [shallow routing](shallow-routing).
+`goto(...)` no longer accepts external URLs. To navigate to an external URL, use `window.location = url`. The `state` object now contributes to `$page.state` and requires an `App.PageState` shape, see [shallow routing](shallow-routing) for more info.
```

Comment:
```suggestion
`goto(...)` no longer accepts external URLs. To navigate to an external URL, use `window.location = url`. The `state` object now contributes to `$page.state` and requires an `App.PageState` shape. See [shallow routing](shallow-routing) for more details.
```

Reply:
'contribute' suggests the values are merged or something. also, `state` can be any object at all if you haven't specified a `PageState` interface

```suggestion
`goto(...)` no longer accepts external URLs. To navigate to an external URL, use `window.location = url`. The `state` object now determines `$page.state` and must adhere to the `App.PageState` interface, if declared. See [shallow routing](shallow-routing) for more details.
```